src/features/command-palette/command-palette.ts

```diff
@@ -0,0 +1,180 @@
+// this is a workaround to still get the function including type support
+const configureAutocomplete: (settings: AutocompleteSettings<LabeledAction>) => AutocompleteResult = require("autocompleter");
+
+@injectable()
+export class CommandPalette extends AbstractUIExtension {
+    static readonly ID = "command-palette";
+    readonly id = CommandPalette.ID;
+
+    readonly containerClass = "command-palette";
+    readonly xOffset = 20;
+    readonly yOffset = 30;
+    readonly defaultWidth = 400;
+
+    protected inputElement: HTMLInputElement;
+    protected autoCompleteResult: AutocompleteResult;
+
+    protected paletteContext?: CommandExecutionContext;
+    protected contextActions?: LabeledAction[];
+
+    constructor(
+        @inject(TYPES.ViewerOptions) protected options: ViewerOptions,
+        @inject(TYPES.IActionDispatcherProvider) protected actionDispatcherProvider: IActionDispatcherProvider,
+        @inject(TYPES.ICommandPaletteActionProviderRegistry) protected actionProvider: ICommandPaletteActionProviderRegistry,
+        @inject(TYPES.ILogger) protected logger: ILogger) {
+        super(options, logger);
+    }
+
+    show(context: CommandExecutionContext) {
+        super.show(context);
+        if (this.inputElement.value) {
+            this.inputElement.setSelectionRange(0, this.inputElement.value.length);
+        }
+        this.autoCompleteResult = configureAutocomplete(this.autocompleteSettings(context));
+        this.inputElement.focus();
+    }
+
+    protected initializeUIExtension(containerElement: HTMLElement) {
+        containerElement.style.position = "absolute";
+        this.inputElement = document.createElement('input');
+        this.inputElement.style.width = '100%';
+        containerElement.appendChild(this.inputElement);
+        this.inputElement.onblur = () => window.setTimeout(() => this.hide(), 200);
+    }
+
+    protected updateBeforeBecomingVisible(containerElement: HTMLElement, context: CommandExecutionContext) {
+        let x = this.xOffset;
+        let y = this.yOffset;
+        const selectedElements = toArray(context.root.index.all().filter(e => isSelectable(e) && e.selected));
+        if (selectedElements.length === 1) {
+            const firstElement = selectedElements[0];
+            if (isBoundsAware(firstElement)) {
+                const viewport = findParentByFeature(firstElement, isViewport);
+                if (viewport) {
+                    x += (firstElement.bounds.x - viewport.scroll.x) * viewport.zoom;
+                    y += (firstElement.bounds.y - viewport.scroll.y) * viewport.zoom;
+                } else {
+                    x += firstElement.bounds.x;
+                    y += firstElement.bounds.y;
+                }
+            }
+        }
+        containerElement.style.left = `${x}px`;
+        containerElement.style.top = `${y}px`;
+        containerElement.style.width = `${this.defaultWidth}px`;
```

Comment:
The input box should appear at the side of the selected element, not on top of it.